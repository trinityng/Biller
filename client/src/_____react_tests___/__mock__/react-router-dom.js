import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;