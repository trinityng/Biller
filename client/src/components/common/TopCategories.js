import React from 'react';
import PropTypes from 'prop-types';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  CardFooter,
  Row,
  Col,
  FormSelect,
} from 'shards-react';

const TopCategories = props => (
  <Card small>
    <CardHeader className="border-bottom">
      <h6 className="m-0">Categories</h6>
      <div className="block-handle" />
    </CardHeader>

    <CardBody className="p-0">
      <ListGroup small flush className="list-group-small">
        {props.category_expense.map((item, idx) => (
          <ListGroupItem key={idx} className="d-flex px-3">
            <span className="text-semibold text-fiord-blue">
              {item.category}
            </span>
            <span className="ml-auto text-right text-semibold text-reagent-gray">
              ${item.total.toFixed(0)}
            </span>
          </ListGroupItem>
        ))}
      </ListGroup>
    </CardBody>

    <CardFooter className="border-top">
      <Row>
        {/* Time Span */}
        <Col>
          <FormSelect
            size="sm"
            value="last-week"
            style={{ maxWidth: '130px' }}
            onChange={() => {}}
          >
            <option value="last-week">Last Week</option>
            <option value="today">Today</option>
            <option value="last-month">Last Month</option>
            <option value="last-year">Last Year</option>
          </FormSelect>
        </Col>

        {/* View Full Report */}
        <Col className="text-right view-report">
          {/* eslint-disable-next-line */}
          <a href="#">Full report &rarr;</a>
        </Col>
      </Row>
    </CardFooter>
  </Card>
);

TopCategories.propTypes = {
  /**
   * The component's title.
   */
  title: PropTypes.string,
  /**
   * The referral data.
   */
  referralData: PropTypes.array,
};

TopCategories.defaultProps = {
  title: 'Top Categories',
  referralData: [
    {
      title: 'Shopping',
      value: '$1,291',
    },
    {
      title: 'Travel',
      value: '$1,291',
    },
    {
      title: 'Food',
      value: '$1,291',
    },
    {
      title: 'Utilities',
      value: '$1,291',
    },
  ],
};

export default TopCategories;
