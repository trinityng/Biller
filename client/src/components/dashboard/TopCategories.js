import React from 'react';
import PropTypes from 'prop-types';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  Badge,
} from 'shards-react';

const TopCategories = props => (
  <Card small>
    <CardHeader className="border-bottom">
      <h6 className="m-0">Categories</h6>
      <div className="block-handle" />
    </CardHeader>

    <CardBody className="p-0">
      <ListGroup small flush className="list-group-small">
        {props.category_expense.map((item, idx) => (
          <ListGroupItem key={idx} className="d-flex px-3">
            <span className="text-semibold text-fiord-blue">
              {item.category}
            </span>
            <Badge
              theme="info"
              className="mb-0 mr-0 ml-auto text-right text-semibold "
            >
              ${item.total.toFixed(0)}
            </Badge>
          </ListGroupItem>
        ))}
      </ListGroup>
    </CardBody>
  </Card>
);

TopCategories.propTypes = {
  /**
   * The component's title.
   */
  title: PropTypes.string,
  /**
   * The referral data.
   */
  referralData: PropTypes.array,
};

TopCategories.defaultProps = {
  title: 'Top Categories',
  referralData: [
    {
      title: 'Shopping',
      value: '$1,291',
    },
    {
      title: 'Travel',
      value: '$1,291',
    },
    {
      title: 'Food',
      value: '$1,291',
    },
    {
      title: 'Utilities',
      value: '$1,291',
    },
  ],
};

export default TopCategories;
