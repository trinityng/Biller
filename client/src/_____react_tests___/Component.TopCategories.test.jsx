import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Router } from "react-router-dom";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import TopCategories from '../components/dashboard/TopCategories';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});

describe('TopCategories components', () => {
    let initialState = {};
    let store;

    beforeEach(() => {
        store = mockStore(initialState)
    });

    it(' should render TopCategories correctly', () => {
        const wrapper = mount(
            <TopCategories 
            category_expense = {[  ]}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});