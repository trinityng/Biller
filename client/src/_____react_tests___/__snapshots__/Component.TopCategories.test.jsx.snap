// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopCategories components  should render TopCategories correctly 1`] = `ShallowWrapper {}`;
